psi4/src/psi4/libmints/wavefunction.cc

```diff
@@ -235,6 +235,110 @@ void Wavefunction::deep_copy(const Wavefunction *other)
 
 }
 
+void Wavefunction::c1_deep_copy(SharedWavefunction other)
+{
+    c1_deep_copy(other.get());
+}
+
+void Wavefunction::c1_deep_copy(const Wavefunction *other)
+{
+    if (!other->S_) {
+        throw PSIEXCEPTION("Wavefunction::c1_deep_copy must copy an initialized wavefunction.");
+    }
+
+    /// From typical constructor
+    /// Some member data is not clone-able so we will copy
+    name_ = other->name_;
+    molecule_ = std::shared_ptr<Molecule>(new Molecule(other->molecule_->clone()));
+    molecule_->reset_point_group("c1");
+    molecule_->set_orientation_fixed(true);
+    molecule_->set_com_fixed(true);
+    molecule_->update_geometry();
+
+    basisset_ = other->basisset_; // CDS: think we need to reset this
+                                  // b/c it contains a molecule
+    basissets_ = other->basissets_; // Still cannot copy basissets
```

Comment:
As it stand this basisset still has a pointer to the old molecule with symmetry. In this particular case we could just add a `molecule` setter to the BasisSet object as the previous wavefunction will be destroyed. This wont work in the general case as it would screw up anything downstream that used the previous Wavefunction. We could consider making the `to_c1` an attribute so that this call is a *destructive* process (just updates the current Wavefunction). This could be a elegant way around this particular issue until the BasisSet object is cleaned up a bit. I think @andysim is working on some parts of this.